### PR TITLE
New version: Logitech.GHUB version 2023.6.430723

### LINKED ISSUE #113476: [Package Issue]: Logitech.GHUB package fails to install with "Installer hash does not match."
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installer hash mismatch.

### Brief description of your issue

The Logitech G Hub software (Id: Logitech.GHUB, Version: 2023.5.418755) fails to install with an error message of "Installer hash does not match."

### Steps to reproduce

Run `winget install --id Logitech.GHUB`

### Actual behavior

Install fails:

`Downloading https://download01.logi.com/web/ftp/pub/techsupport/gaming/lghub_installer.exe
  ██████████████████████████████  39.6 MB / 39.6 MB
Installer hash does not match.`

### Expected behavior

Install succeeds

### Environment

```raw
Windows: Windows.Desktop v10.0.22621.1992
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.20.1881.0
```

### Screenshots and Logs

_No response_

### PR DESCRIPTION
**Pull request by RobertHarnett**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

fixes #113476

-----

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-pkgs/pull/113552&drop=dogfoodAlpha

### CODE REVIEW

File: manifests/l/Logitech/GHUB/2023.6/Logitech.GHUB.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Created using wingetcreate 1.2.8.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: Logitech.GHUB
+PackageVersion: 2023.6
```
Comment (Trenly) [inline manifests/l/Logitech/GHUB/2023.6/Logitech.GHUB.installer.yaml:5]:
```suggestion
PackageVersion: 2023.6.430723
```
This should match control panel

File: manifests/l/Logitech/GHUB/2023.6/Logitech.GHUB.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Created using wingetcreate 1.2.8.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: Logitech.GHUB
+PackageVersion: 2023.6
```
Comment (Trenly) [inline manifests/l/Logitech/GHUB/2023.6/Logitech.GHUB.locale.en-US.yaml:5]:
```suggestion
PackageVersion: 2023.6.430723
```
This should match control panel

File: manifests/l/Logitech/GHUB/2023.6/Logitech.GHUB.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Created using wingetcreate 1.2.8.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.4.0.schema.json
+
+PackageIdentifier: Logitech.GHUB
+PackageVersion: 2023.6
```
Comment (Trenly) [inline manifests/l/Logitech/GHUB/2023.6/Logitech.GHUB.yaml:5]:
```suggestion
PackageVersion: 2023.6.430723
```
This should match control panel

**Review by Trenly — CHANGES_REQUESTED** [review]
Please also remove the previous version as it is no longer available

**Comment by RobertHarnett** [comment]
Updated version number and removed old version as required.

**Review by Trenly — APPROVED** [review]

**Pull request merged**